bibtex-actions.el

```diff
@@ -225,53 +244,75 @@ offering the selection candidates"
            bibtex-completion-bibliography)))
     (seq-difference local-bib-files global-bib-files)))
 
+(defun bibtex-actions-get-value (field item &optional _default)
+  "Return biblatex FIELD value for ITEM."
+  (or (cdr (assoc-string field item 'case-fold))
+      (cl-loop for fname in (cdr (assoc field bibtex-actions-field-map))
+               when (cdr (assoc-string fname item 'case-fold))
+                         return (cdr (assoc-string fname item 'case-fold)))
+      ""))
+
+ (defun bibtex-actions-shorten-names (names)
+  "Return a list of family names from a list of full NAMES.
+
+To better accomomodate corporate names, this will only shorten
+personal names of the form 'family, given'."
+  (mapconcat
+   (lambda (name)
+     (if (eq 1 (length name))
+         (cdr (split-string name " "))
+       (car (split-string name ", "))))
+   (split-string names " and ") ", "))
+
 (defun bibtex-actions--format-candidates (&optional context)
   "Format candidates, with optional hidden CONTEXT metadata.
 This both propertizes the candidates for display, and grabs the
 key associated with each one."
-  (let* ((main-template
+  (let* ((candidates
+          (ht-values (parsebib-parse bibtex-completion-bibliography)))
```

Comment:
I think it would be better if this function takes a list of files as a mandatory arguments instead of relying on  `bibtex-completion-bibliography`. I think the only place this will need to change is `bibtex-actions-refresh` where we can just `bibtex-completion-bibliography` as the argument for local files and similarly deal with the local files without having to let-bind `bibtex-completion-bibliography` which I find somewhat clumsy.

Reply:
Yeah, I was thinking about this.